### PR TITLE
Always add a hash to the name of the files created by the default assets loaders

### LINKED ISSUE #73: File-loader not works with different images with same name
Hi.
I've 2 different images with same name:
```
assets/images/some.png
assets/another_images/some.png
```
In scss file i use both. File-loader copy only first image and skips second.
I have to rewrite the rule in config-generator.js to:
```
{
                test: /\.(png|jpg|jpeg|gif|ico|svg)$/,
                loader: 'file-loader',
                options: {
                    name: `images/[name]_[md5:hash:hex:3]${this.webpackConfig.useVersioning ? '.[hash]' : ''}.[ext]`,
                    publicPath: this.webpackConfig.getRealPublicPath()
                }
            }
```

But is this a right way ?
Second question: how can i overwrite that rule in webpack.config.js ?

### PR DESCRIPTION
**Pull request by Lyrkan**
This PR modifies the configuration of the default assets loaders so a `[hash]` is always added to the output file names, even if versioning is disabled.

It basically prevents having two files with the same name (but in a different directory) overwriting each other during build.

This closes #73 and was further discussed in #103.

### CODE REVIEW

File: lib/config-generator.js [hunk 1]
```diff
@@ -127,15 +127,15 @@ class ConfigGenerator {
                 test: /\.(png|jpg|jpeg|gif|ico|svg)$/,
                 loader: 'file-loader',
                 options: {
-                    name: `images/[name]${this.webpackConfig.useVersioning ? '.[hash]' : ''}.[ext]`,
+                    name: 'images/[name].[hash].[ext]',
```
Comment (weaverryan) [inline lib/config-generator.js]: What do you think about using a shorter hash - like `[hash:8]`? The longer seems overkill... I should have done this originally.

**Comment by weaverryan** [comment]
One minor comment - but this looks great - really nice job with the test!

File: lib/config-generator.js (hunk 1 above)
Comment (Lyrkan) [inline lib/config-generator.js]: Done, no idea what's up with AppVeyor though (same thing for #103)

**Comment by weaverryan** [comment]
Thank you @Lyrkan! Test failure is from master and unrelated - I'll fix it :)

**Pull request merged**